docs/source/development/index.rst

```diff
@@ -23,82 +23,77 @@ For an overview of core StackStorm code structure, please refer to
-  documentation and examples.
+* Any non-trivial change must contain corresponding tests. For more information, refer to the
+  :doc:`Testing page</development/testing>` (or the :doc:`Pack Testing</development/pack_testing>`
+  page for pack development).
+* All the functions and methods must contain Sphinx docstrings which are used to generate the API
+  documentation. We follow the Apache Libcloud project docstrings conventions. For more
+  information, refer to the `Docstring conventions`_ page.
+* If you are adding a new feature, make sure to add corresponding documentation and examples.
 
 Code style guide
 ----------------
 
-* We follow `PEP8 Python Style Guide`_
-* Use 4 spaces for a tab
-* Use 100 characters in a line
-* Make sure edited file doesn't contain any trailing whitespace
-* Make sure that all the source files contains an Apache 2.0 license header.
-  For example, see one of the existing Python files with source code.
-* You can verify that your modifications don't break any rules by running the
-  lint script - ``make flake8``
+* We follow `PEP8 Python Style Guide`_.
+* Use 4 spaces for a tab.
+* Use 100 characters in a line.
+* Make sure edited files don't contain any trailing whitespace.
+* Make sure that all the source files contain an Apache 2.0 license header. For an example, see one
+  of the existing Python files.
+* You can verify that your modifications don't break any rules by running the lint script -
+  ``make flake8``
 
-Most StackStorm repositories use shared Flake8 and PyLint configuration files
-which you can get from the `lint-configs repo <https://github.com/StackStorm/lint-configs>`_.
+Most |st2| repositories use shared Flake8 and PyLint configuration files, which you can get from
+the `lint-configs repo <https://github.com/StackStorm/lint-configs>`_.
 
-And most importantly, follow the existing style in the file you are editing and
-**be consistent**.
+And most importantly, follow the existing style in the file you are editing and **be consistent**.
 
 General coding guidelines
 -------------------------
 
 Logging
 ~~~~~~~
 
-Logging is important because it increases the visibility and makes the project
-easiest to debug and support.
+Logging is important because it increases the visibility and makes the project easier to debug
+and support.
 
-You are encouraged to generously use the log statements across the code base -
-you should log every event which increases the visibility and / or makes the
-product easier to debug and support.
+You are encouraged to generously use the log statements across the code base. You should log every
+event which increases the visibility and/or makes the product easier to debug and support.
 
-Every log statement should also include as much as useful additional context as
-possible. This context should be included in the dictionary which is passed via
-``extra`` keyword argument to the logger method as shown bellow.
```

Comment:
I'm sure all those `bellow` occurrences were me :trollface: 

Accidentally added `bellow` as a correct spelling to my vim dictionary, but I fixed that like a year ago ;P

Reply:
Ha! Every time I think I've got rid of them all, another one pops up. I think there's a few in the docstrings too